pkg/scan/scan.go

```diff
@@ -44,6 +46,7 @@ func analyzeLayers(storage database.Datastore, registry types.Registry, image *t
 	for _, layer := range layers {
 		layerReadCloser := &LayerDownloadReadCloser{
 			Downloader: func() (io.ReadCloser, error) {
+				// It is assumed layer is a valid digest.Digest, so no need to parse and validate here.
```

Comment:
I am curious: Why you did not validate anyway to be on the safe side?

Reply:
Same for `handleManifest`.

Reply:
I originally validated here, but I removed it once I (believed) it was accounted for elsewhere. But you're right. Best to be safe

Reply:
For `handleManifest`, `ref` may not be a digest, but rather a tag, so I didn't want to do the check there